warehouse/core/src/main/java/datawave/util/AccumuloCounterSource.java

```diff
@@ -62,9 +62,8 @@ public String getNextIdentifier() {
     public boolean hasNext() {
         
         if (null == iterator) {
-            try {
-                Authorizations auths = connector.securityOperations().getUserAuthorizations(username);
-                BatchScanner scanner = connector.createBatchScanner(queryTable, auths, 100);
+            try (BatchScanner scanner = connector.createBatchScanner(queryTable, connector.securityOperations().getUserAuthorizations(username), 100)) {
```

Comment:
I'm not sure what is the right answer here, but I'm pretty sure this change breaks this class. You will close the batch scanner in the hasNext call here, however the side effect is to retrieve an iterator from the batch scanner and use it in the nextIterator call that is made on the current and future hasNext calls. However, the iterator will be invalid since the underlying batch scanner has been closed.

Reply:
Fixed in next commit

Reply:
How was this fixed in the next commit?

Reply:
what, you want me to actually PUSH the commit too?

Oops. Pushed it now.